trajopt/include/trajopt/file_write_callback.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <trajopt/common.hpp>
+#include <trajopt/problem_description.hpp>
+#include <trajopt_utils/eigen_conversions.hpp>
+#include <tesseract_core/basic_kin.h>
+#include <tesseract_core/basic_env.h>
+#include <trajopt/plot_callback.hpp>
+#include <functional>
+#include <set>
+#include <fstream>
+#include <string>
+#include <stdio.h>
+#include <Eigen/Core>
```

Comment:
Don't put includes in a public header that aren't needed _in the header_. Move as much to the source file as possible.